source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -67,8 +67,12 @@ void UdpStatsdSink::flush(Stats::MetricSnapshot& snapshot) {
 }
 
 void UdpStatsdSink::onHistogramComplete(const Stats::Histogram& histogram, uint64_t value) {
-  // For statsd histograms are all timers.
-  const std::string message(fmt::format("{}.{}:{}|ms{}", prefix_, getName(histogram),
+  // For statsd histograms are all timers, append the ASCII SI symbol to disambiguate.
+  std::string si_name = histogram.unitSymbol().length() > 0
+                            ? fmt::format("{}_{}", getName(histogram), histogram.unitSymbol())
+                            : getName(histogram);
+
+  const std::string message(fmt::format("{}.{}:{}|ms{}", prefix_, si_name,
```

Comment:
why do we have hardcoded "ms" here *and* use unitSymbol() above?

Should we just ASSERT that the units is milliseconds here?

Or add comments to explain the strategy here.

Reply:
statsd uses "ms" to specify the metric is a histogram, under the hood it does not have histogram vs timer distinction and everything is a timer. Added comments to disambiguate.